configuration/pih/liquibase/sql/create_functions.sql

```diff
@@ -1945,3 +1945,131 @@ limit 1;
 
 END
 #
+-- The following diagnosis functions can work together to provide all of the details of diagnoses entered in encounters
+-- they each accept encounter_id and offset and will return the details of the nth (based on offset) encounter,
+-- ordered by: 
+--   primary diagnoses first, then other diagnoses with an order, then diagnoses without an order. obs_id is included as a final sort factor
+#
+-- The following function accepts encounter_id and offset
+-- It will return the obs_group_id of the construct of the nth diagnosis of that encounter (n being the offset)
+#
+DROP FUNCTION IF EXISTS diagnosis_obs_group_id;
+#
+CREATE FUNCTION diagnosis_obs_group_id(_encounter_id int(11), _offset int(11)) RETURNS int(11)
+    DETERMINISTIC
+BEGIN
+
+DECLARE ret int(11);
+
+select ogr.obs_id into ret from obs ogr 
+ left outer join obs oo on oo.obs_group_id = ogr.obs_id and oo.concept_id = concept_from_mapping('PIH','7537') and oo.voided = 0
+where ogr.encounter_id = _encounter_id
+and ogr.concept_id = concept_from_mapping('PIH','7539')
+and ogr.voided = 0
+order by ISNULL(oo.value_coded) ASC ,LOCATE('7534',retrieveConceptMapping(oo.value_coded,'PIH')) DESC, ogr.obs_id ASC
```

Comment:
So I guess this works if there are exactly 2 choices for order - primary and secondary - but if additional options were available we'd have to tweak it, right?